code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -574,3 +574,198 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+	sleep(5)
+	playsound(loc, 'sound/effects/pop_expl.ogg', 50, 1)
+	add_fingerprint(user)
+	if (H && !qdeleted(H))
+		if (B && !qdeleted(B))
+			H.internal_organs -= B
+			qdel(B)
+		gibs(H.loc, H.viruses, H.dna)
+		return (BRUTELOSS)
+	qdel(src)
+
+
+/obj/item/clothing/mask/vape/New(loc, var/param_color = null)
+	..()
+	create_reagents(chem_volume)
+	reagents.set_reacting(FALSE) // so it doesn't react until you light it
+	reagents.add_reagent("nicotine", 50)
+	if(!icon_state)
+		if(!param_color)
+			param_color = pick("red","blue","black","white","green","purple","yellow","orange")
+		icon_state = "[param_color]_vape"
+		item_state = "[param_color]_vape"
+
+//The reagent thing was here
+/obj/item/clothing/mask/vape/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/weapon/screwdriver))
+		if(!screw)
+			screw = 1
+			flags |= OPENCONTAINER
+			user << "<span class='notice'>You open the cap on the [src].</span>"
+			if(super & !emagged)
+				var/image/I = (image(icon, "vapeopen_med"))
+				overlays += I
+			if(emagged)
+				var/image/I = (image(icon, "vapeopen_high"))
+				overlays += I
+			if(!super & !emagged)
+				var/image/I = (image(icon, "vapeopen_low"))
+				overlays += I
+		else
+			screw = 0
+			flags &= ~OPENCONTAINER
+			user << "<span class='notice'>You close the cap on the [src].</span>"
+			overlays.Cut()
+
+	if(istype(O, /obj/item/device/multitool))
+		if(screw && !emagged)//also kinky
+			if(!super)
+				overlays.Cut()
+				super = 1
+				user << "<span class='notice'>You increase the voltage in the [src].</span>"
+				var/image/I = (image(icon, "vapeopen_med"))
+				overlays += I
+			else
+				overlays.Cut()
+				super = 0
+				user << "<span class='notice'>You decrease the voltage in the [src].</span>"
+				var/image/I = (image(icon, "vapeopen_low"))
+				overlays += I
+
+		if(screw && emagged)
+			user << "<span class='notice'>The [name] can't be modified!</span>"
+
+
+/obj/item/clothing/mask/vape/emag_act(mob/user)
+	if(screw)
+		if(!emagged)
+			overlays.Cut()
+			emagged = 1
+			super = 0
+			user << "<span class='warning'>You maximize the voltage in the [src]!</span>"
+			var/image/I = (image(icon, "vapeopen_high"))
+			overlays += I
+			var/datum/effect_system/spark_spread/sp = new /datum/effect_system/spark_spread //for effect
+			sp.set_up(5, 1, src)
+			sp.start()
+		else
+			user << "<span class='warning'>The [name] is already emagged!</span>"
+	else
+		user << "<span class='notice'>You need to open the cap to do that!</span>"
+
+/obj/item/clothing/mask/vape/attack_self(mob/user)
+	if(reagents.total_volume > 0)
+		user << "<span class='notice'>You empty [src] of all reagents.</span>"
+		reagents.clear_reagents()
+	return
+
+/obj/item/clothing/mask/vape/equipped(mob/user, slot)
+	if(slot == slot_wear_mask)
+		if(!screw)
+			user << "<span class='notice'>You start ripping fat vapes with [src].</span>"
+			reagents.set_reacting(TRUE) // allowing reagents to react after being lit
+			SSobj.processing |= src
```

Comment:
This is deprecated. You need to call START_PROCESSING(SSobj, src) and STOP_PROCESSING(SSobj, src) accordingly.